deploy/elasticsearch/templates/clusterdefinition.yaml

```diff
@@ -0,0 +1,223 @@
+              value: "no"
+            - name: ELASTICSEARCH_NODE_ROLES  # node type when behaving as a 'dedicated node'
+              value: ""  
+            - name: ELASTICSEARCH_TRANSPORT_PORT_NUMBER
+              value: "9300"
+            - name: ELASTICSEARCH_HTTP_PORT_NUMBER
+              value: "9200"
+            - name: ELASTICSEARCH_CLUSTER_HOSTS
+              value: ""
+            - name: ELASTICSEARCH_TOTAL_NODES
+              value: "1"
+            - name: ELASTICSEARCH_CLUSTER_MASTER_HOSTS
+              value: "$(KB_CLUSTER_NAME)-$(KB_COMP_NAME)-0"
+            - name: ELASTICSEARCH_MINIMUM_MASTER_NODES
+              value: "1"
+            - name: ELASTICSEARCH_ADVERTISED_HOSTNAME
+              value: "$(KB_CLUSTER_NAME)-$(KB_COMP_NAME)"
+            - name: ELASTICSEARCH_HEAP_SIZE
+              value: "128m"
+
+            volumeMounts:
+              - mountPath: /usr/share/elasticsearch/data
+                name: data
+              - mountPath: /usr/share/elasticsearch/config/elasticsearch.yaml
+                subPath: elasticsearch.yaml
+                name: elasticsearch-config
+
+# https://discuss.elastic.co/t/kibana-not-able-to-connect-to-elastic-master-in-kubernetes-from-elastic-helm/185759/5
+          - name: elasticsearch-master-graceful-termination-handler
+            imagePullPolicy: IfNotPresent
+            command:
+            - "sh"
+            - -c
+            - |
+              #!/usr/bin/env bash
+              set -e
+
+              http () {
+                  local path="${1}"
+                  if [ -n "${USERNAME}" ] && [ -n "${PASSWORD}" ]; then
+                    BASIC_AUTH="-u ${USERNAME}:${PASSWORD}"
+                  else
+                    BASIC_AUTH=''
+                  fi
+                  curl -XGET -s -k --fail ${BASIC_AUTH} https://$(KB_CLUSTER_NAME)-$(KB_COMP_NAME)-headless:9200:${path}
+              }
+
+              cleanup () {
+                while true ; do
+                  local master="$(http "/_cat/master?h=node" || echo "")"
+                  if [[ $master == "$(KB_CLUSTER_NAME)-$(KB_COMP_NAME)"* && $master != "${NODE_NAME}" ]]; then
+                    echo "This node is not master."
+                    break
+                  fi
+                  echo "This node is still master, waiting gracefully for it to step down"
+                  sleep 1
+                done
+
+                exit 0
+              }
+
+              trap cleanup TERM
+
+              sleep infinity &
+              wait $!
+
+    - name: metrics
+      monitor:
+        builtIn: false
+        exporterConfig:
+          scrapePath: /metrics
+          scrapePort: 9114 # Source: elasticsearch/templates/metrics/deployment.yaml
+      characterType: elasticsearch-metrics
+      workloadType: Stateful
+      service:
+        ports:
+        - name: http-metrics
+          port: 9114
+          targetPort: metrics
```

Comment:
It is better to ensure consistency between the service port and targetPort names.